test/IsolatedEntities/Common/TestRunner.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace IsolatedEntities;
+
+internal static class TestRunner
+{
+    public static async Task<string> RunAsync(TestContext context, string? filter = null, bool listOnly = false)
```

Comment:
It has to run inside a function app. I had no appetite for spending a lot of time figuring out how to get a unit testing framework running inside a function app. Also, my past experience with those frameworks is not great. Takes a lot to get them to do what you want, even if what you want is rather simple. As demonstrated with this little bit of code here.

Reply:
Yeah, I am not sure we should be adding our own test runner. What I recommend is using xunit test fixtures. You can encapsulate starting the function app via this fixture and then author tests which pull in and start the fixture (this starting the function app), and then dispatch HTTP to the function app to run a given scenario.

Reply:
If I understand what you are suggesting, we would *additionally* create a separate xunit test for each test and then have that test invoke the app's unit test via http call?  

Would that be a separate project (given that the function app is a special kind of project)? Would there be any way to avoid having to redefine all tests in both projects?